src/home_robot_sim/home_robot_sim/env/habitat_objectnav_env/visualizer.py

```diff
@@ -195,6 +195,51 @@ def get_semantic_vis(self, semantic_map, rgb_frame=None):
 
         return semantic_map_vis
 
+    def flatten_instance_map(self, instance_map):
+        """
+        Flatten the instance map.
+
+        Args:
+            instance_map: np.ndarray of shape [num_sem_categories - 2, H, W] where each channel has instances labeled as 1, 2, ...
+
+        Returns:
+            instance_map_combined: Flattened instance map with globally combined instance labels.
+            instances_per_category: Number of instances per category.
+        """
+        num_channels, height, width = instance_map.shape
+        instance_map_flattened = instance_map.reshape(num_channels, -1)
+        instances_per_category = np.max(instance_map_flattened, axis=1).astype(np.int64)
+
+        instance_map_combined = instance_map[0].copy()
+
+        if num_channels > 1:
+            cumulative_instances = np.cumsum(instances_per_category[:-1])
+            instance_map_combined += np.sum(
+                instance_map[1:] * cumulative_instances[:, np.newaxis, np.newaxis],
+                axis=0,
+            )
+
+        return instance_map_combined, instances_per_category
+
+    def replace_semantic_palette_with_instance_palette(
+        self,
+        input_colors: List[int],
+        num_instances: List[int],
+        delta: Optional[int] = 20,
+    ) -> List[int]:
```

Comment:
please document what is going on here. how would i use this? what is delta doing?

Reply:
delta is actually pretty confusing to me, not immediately obvious even after looking at the code why this is a param

Reply:
ah i see, delta is a step size through color space... it's a bit odd but i get it

Reply:
Removed this function now. The colors were too close.